internal/impl/devbox.go

```diff
@@ -430,15 +430,45 @@ func (d *Devbox) Exec(cmds ...string) error {
 	return nix.Exec(nixDir, cmds, env)
 }
 
-func (d *Devbox) PluginEnv() (string, error) {
-	pluginEnvs, err := plugin.Env(d.packages(), d.projectDir)
+func (d *Devbox) PrintEnv() (string, error) {
+	script := ""
+	if featureflag.UnifiedEnv.Disabled() {
+		envs, err := plugin.Env(d.packages(), d.projectDir)
+		if err != nil {
+			return "", err
+		}
+		for k, v := range envs {
+			script += fmt.Sprintf("export %s=%s\n", k, v)
+		}
+		return script, nil
+	}
+	envs, err := d.computeNixEnv()
+	// exclude variables that direnv can't process
+	varsToExclude := []string{
+		"buildPhase",
+		"shellHook",
+	}
```

Comment:
This feels a bit brittle in that we may modify computeNixEnv to add other variables which may not work.

In https://github.com/jetpack-io/devbox/pull/634, @ipince added a parameter for shellHook
Do we need to either add another parameter for buildPhase, or for the general concept of `isInteractive` ?

Reply:
Hmmm.... well, ideally we should actually _execute_ `shellHook` in order to have a properly configured environment. Luckily right now, the shellHook when using unified-env is a no-op... but the moment it's not, then this (i.e., the direnv integration) will break.

Maybe that moment will never happen though--it's hard to tell.